staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -394,7 +393,7 @@ func TestWatch(t *testing.T) {
 	// Inject one list error to make sure we test the relist case.
 	etcdStorage = &injectListError{errors: 1, Interface: etcdStorage}
 	defer server.Terminate(t)
-	cacher, _, err := newTestCacher(etcdStorage, 3) // small capacity to trigger "too old version" error
+	cacher, _, err := newTestCacher(etcdStorage) // cache expand with dynamic size, no "too old version" error
```

Comment:
hmm - it seems that the test is no longer testing what it was supposed to test.
This should be fixed to test what we were previously doing (if we broke it over time).

Reply:
Yes, I have broken it in first pr, as it's more of an integration test. And I haven't come up with a very good method to shrink cache and make resource version 1 `Gone` here.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go#L447

Reply:
We should be able to emulate out-of-window by simply using fake clock, right?

Reply:
> Yes, I have broken it in first pr, as it's more of an integration test. And I haven't think of a very good method to shrink cache and make resource version 1 `Gone` here.

Updated watch cache not full and won't shrink. Shall I full it?

Reply:
If this is needed to restore to the original test - yes.

Reply:
OK, I'll try to resolve it tomorrow in my time zone :)